routes/router.js

```diff
@@ -363,499 +363,153 @@ router.route('/about')
-                                                    var imgPath = 'exports/xl/media/' + image; //This is undefined.
-                                                    var A = storedImages;
-                                                    var a = new A;
-                                                    a.projectID = IdOfProject;
-                                                    a.neumeID = neume._id;
-                                                    a.img.data = fs.readFileSync(imgPath);
-                                                    a.img.contentType = 'image/png';
-                                                    a.imgBase64 = a.img.data.toString('base64');
-
-                                                    imageData.push(a.img.data.toString('base64')); //This works for all the images stored in the database.
-                                                    //logger.info(imageData); //This works
-                                                    mongoose.model('neume').find({
-                                                        _id: neume._id
-                                                    }).update({
-                                                            //push the neumes into the imagesBinary array
-                                                            imagePath: 'exports/xl/media/' + neume.imageMedia,
-                                                            imagesBinary: fs.readFileSync('exports/xl/media/' + neume.imageMedia).toString('base64')
-                                                        },
-
-                                                        function(err, data) {
-                                                            if (err) {
-                                                                return logger.error(res, err);
-                                                            }
-                                                            //logger.info(err, data);
-                                                            imageData = [];
-
-                                                            a.save(function(err, a) {
-                                                                if (err) {
-                                                                    return logger.error(res, err);
-                                                                }
-                                                                logger.info('saved img to mongo');
-                                                            });
-                                                        });
-                                                }
-
-                                            })
+                // somtimes the images aren't done extracting before we try to read them
+                // 500ms timeout seems to work for files with up to 100 images
+                setTimeout(function() {
```

Comment:
A fixed timeout period for IO is not ideal. I'd suggest using [promises](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Using_promises) or something similar here.